internal/tools/cloudsql/cloudsqlcloneinstance/cloudsqlcloneinstance.go

```diff
@@ -0,0 +1,187 @@
+)
+
+const kind string = "cloud-sql-clone-instance"
+
+func init() {
+	if !tools.Register(kind, newConfig) {
+		panic(fmt.Sprintf("tool kind %q already registered", kind))
+	}
+}
+
+func newConfig(ctx context.Context, name string, decoder *yaml.Decoder) (tools.ToolConfig, error) {
+	actual := Config{Name: name}
+	if err := decoder.DecodeContext(ctx, &actual); err != nil {
+		return nil, err
+	}
+	return actual, nil
+}
+
+// Config defines the configuration for the clone-instance tool.
+type Config struct {
+	Name         string   `yaml:"name" validate:"required"`
+	Kind         string   `yaml:"kind" validate:"required"`
+	Description  string   `yaml:"description"`
+	Source       string   `yaml:"source" validate:"required"`
+	AuthRequired []string `yaml:"authRequired"`
+}
+
+// validate interface
+var _ tools.ToolConfig = Config{}
+
+// ToolConfigKind returns the kind of the tool.
+func (cfg Config) ToolConfigKind() string {
+	return kind
+}
+
+// Initialize initializes the tool from the configuration.
+func (cfg Config) Initialize(srcs map[string]sources.Source) (tools.Tool, error) {
+	rawS, ok := srcs[cfg.Source]
+	if !ok {
+		return nil, fmt.Errorf("no source named %q configured", cfg.Source)
+	}
+	s, ok := rawS.(*cloudsqladmin.Source)
+	if !ok {
+		return nil, fmt.Errorf("invalid source for %q tool: source kind must be `cloud-sql-admin`", kind)
+	}
+
+	allParameters := tools.Parameters{
+		tools.NewStringParameter("projectId", "The project ID of the project containing the instance to be cloned."),
+		tools.NewStringParameter("sourceInstanceName", "The name of the instance to be cloned."),
+		tools.NewStringParameter("destinationInstanceName", "The name of the new instance that will be created by cloning the source instance."),
+		// point in time, preferred zone and preferred secondary zone are optional
+		tools.NewStringParameterWithRequired("pointInTime", "The timestamp in RFC 3339 format to which the source instance should be cloned.", false),
+		tools.NewStringParameterWithRequired("preferredZone", "The preferred zone for the new instance.", false),
+		tools.NewStringParameterWithRequired("preferredSecondaryZone", "The preferred secondary zone for the new instance.", false),
+	}
+	paramManifest := allParameters.Manifest()
+
+	description := cfg.Description
+	if description == "" {
+		description = "Clone an existing Cloud SQL instance into a new instance. Required fields are projectId (the GCP project ID), sourceInstanceName (the name of the instance to clone), and destinationInstanceName (the name to assign to the new instance). Optional fields include pointInTime (a timestamp for point-in-time cloning), preferredZone (the desired zone for the new instance), and preferredSecondaryZone (the preferred secondary zone for high-availability setups). The call returns a Cloud SQL Operation object and then uses the wait_for_operation tool with the multiplier overriden to 4 to poll the opertation status till it is marked DONE."
```

Comment:
This description focuses on the inputs but the inputs are already specified by the tool. Can we iterate on this description to either be shortened to save context token or have specific details to improve usage?